### PR TITLE
Hetman

### PR DESCRIPTION
**Pull request by Hetman18**
Add Auth page objects, rename classes.

### CODE REVIEW

File: src/main/java/page/auth/AuthPage.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    private WebElement signInButton;
+
+    @FindBy(xpath = FORGOT_PASSWORD_LINK_XPATH)
+    private WebElement forgotPasswordLink;
+
+    @FindBy(xpath = REGISTRATION_LINK_XPATH)
+
+    private WebElement registrationLink;
+
+    public Object logInRole(String email, String password, Object newPage) {
```
Comment (OlekLiakh) [inline src/main/java/page/auth/AuthPage.java]: type Page is more suitable for newPage variable and method's return type

File: src/main/java/page/auth/AuthPage.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    private WebElement registrationLink;
+
+    public Page logInRole(String email, String password, Page newPage) {
+        fillField(emailWebElement, email);
+        fillField(passwordInput, password);
+        clickElement(signInButton);
+        return newPage;
+    }
+
+    public AuthPage fillEmailInput(String email) {
```
Comment (fresh-ash) [inline src/main/java/page/auth/AuthPage.java]: Ok

File: src/main/java/page/auth/AuthPage.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        fillField(emailWebElement, email);
+        return this;
+    }
+
+    public Page fillPasswordInputAndSignIn(String password, Page newPage) {
```
Comment (fresh-ash) [inline src/main/java/page/auth/AuthPage.java]: Better, split into two methods: fillField and clickSave. You need these methods for invalid data.

File: src/main/java/page/base/SideBar.java [hunk 4]
```diff
@@ -19,6 +19,11 @@ public WebElement getSideBarItem(SideMenuItem item){
         return getItemFromMenu(sideBarItems, item.itemName);
     }
 
+    public Page clickSideBarItem(SideMenuItem item, Page page){
```
Comment (fresh-ash) [inline src/main/java/page/base/SideBar.java]: For what sends Page into this method?

File: src/main/java/page/courses/PageObject.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
+package page.courses;
```
Comment (fresh-ash) [inline src/main/java/page/courses/PageObject.java]: For what this class?

File: src/main/java/testUI/BaseTest.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+
+import java.time.Duration;
+
+public class BaseTest {
```
Comment (fresh-ash) [inline src/main/java/testUI/BaseTest.java]: Won't we use DriverFactory? That driver and path work ONLY for your local environment. Don't push files like this.

File: src/main/java/page/auth/AuthPage.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    private WebElement signInButton;
+
+    @FindBy(xpath = FORGOT_PASSWORD_LINK_XPATH)
+    private WebElement forgotPasswordLink;
+
+    @FindBy(xpath = REGISTRATION_LINK_XPATH)
+
+    private WebElement registrationLink;
+
+    public Page logInRole(String email, String password, Page newPage) {
```
Comment (fresh-ash) [inline src/main/java/page/auth/AuthPage.java]: You send into this method some Page and return that one. For what?

**Pull request closed without merging**